### PR TITLE
docs: add Vite section to Future Flags docs

### PR DESCRIPTION
**Pull request by brookslybrand**
Closes: #

- [ ] Docs
- [ ] Tests

Testing Strategy:

### CODE REVIEW

**Review by pcattori — APPROVED** [review]

File: docs/start/future-flags.md [hunk 1]
```diff
@@ -17,6 +17,215 @@ First update to the latest minor version of v2.x to have the latest future flags
 npm install @remix-run/{dev,react,node,etc.}@2
 ```
 
+## Vite Plugin
+
+**Background**
+
+Remix no longer uses it's own, closed compiler (now referred to as the "Classic Compiler"), and instead uses [Vite][vite]. Vite is a powerful, performant and extensible development environment for JavaScript projects. [View the Vite docs][vite-docs] for more information on performance, troubleshooting, etc.
```
Comment (markdalgleish) [inline docs/start/future-flags.md:24]:
```suggestion
Remix no longer uses its own, closed compiler (now referred to as the "Classic Compiler"), and instead uses [Vite][vite]. Vite is a powerful, performant and extensible development environment for JavaScript projects. [View the Vite docs][vite-docs] for more information on performance, troubleshooting, etc.
```

File: docs/start/future-flags.md [hunk 2]
```diff
@@ -17,6 +17,215 @@ First update to the latest minor version of v2.x to have the latest future flags
...
+
+While this is not a future flag, new features and some feature flags are only available in the Vite plugin, and the Classic Compiler will be deprecated in the next version of Remix.
```
Comment (markdalgleish) [inline docs/start/future-flags.md:26]:
```suggestion
While this is not a future flag, new features and some feature flags are only available in the Vite plugin, and the Classic Compiler will be removed in the next version of Remix.
```

Comment (pcattori) [reply docs/start/future-flags.md:26]: nice catch 👍

File: docs/start/future-flags.md [hunk 3]
```diff
@@ -17,6 +17,215 @@ First update to the latest minor version of v2.x to have the latest future flags
...
+```
+
+👉 **Update `tsconfig.json`**
+
+Update the `types` field in `tsconfig.json` and make sure `skipLibCheck`, `module`, and `moduleResolution` are all set correctly.
+
+```diff filename=tsconfig.json
+{
+  "compilerOptions": {
++    "types": ["@remix-run/node", "vite/client"],
```
Comment (alcpereira) [inline docs/start/future-flags.md:90]:
```suggestion
+   "types": ["@remix-run/node", "vite/client"],
```

File: docs/guides/vite.md [hunk 4]
```diff
@@ -279,39 +279,34 @@ Vite handles imports for all sorts of different file types, sometimes in ways th
...
-
-👉 **Replace reference to `remix.env.d.ts` with `env.d.ts` in `tsconfig.json`**
+Update the `types` field in `tsconfig.json` and make sure `skipLibCheck`, `module`, and `moduleResolution` are all set correctly.
 
 ```diff filename=tsconfig.json
-- "include": ["remix.env.d.ts", "**/*.ts", "**/*.tsx"],
-+ "include": ["env.d.ts", "**/*.ts", "**/*.tsx"],
+{
+  "compilerOptions": {
++    "types": ["@remix-run/node", "vite/client"],
```
Comment (alcpereira) [inline docs/guides/vite.md:289]:
```suggestion
+   "types": ["@remix-run/node", "vite/client"],
```

File: docs/guides/vite.md [hunk 5]
```diff
@@ -1093,7 +1088,7 @@ Finally, we were inspired by how other frameworks implemented Vite support:
 [vite-config]: ../file-conventions/vite-config
 [vite-plugins]: https://vitejs.dev/plugins
 [vite-features]: https://vitejs.dev/guide/features
-[supported-remix-config-options]: #configuration
+[supported-remix-config-options]: https://remix.run/docs/en/main/file-conventions/vite-config
```
Comment (alcpereira) [inline docs/guides/vite.md:1091]:
```suggestion
[supported-remix-config-options]: ../file-conventions/vite-config
```
This is the same link as `[vite-config]`, is this intended?

Comment (brookslybrand) [reply docs/guides/vite.md:1091]: Ah, thanks for the catch!

**Pull request merged**